### PR TITLE
Add higher-level namespace API

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: lib/namespace.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    return out
+  }
+
+  verify (signature, payload, publicKey) {
+    sodium.crypto_generichash(this._hash, payload)
+    return sodium.crypto_sign_verify_detached(signature, this._signable, publicKey)
+  }
+
+  hash (payload, out = b4a.allocUnsafe(32)) {
+    sodium.crypto_generichash(out, this.ns, payload)
```
Comment (mafintosh) [inline lib/namespace.js:25]: Should be hash([ns, payload])

File: lib/namespace.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    this.ns = ns
+
+    this._signable = b4a.allocUnsafe(64)
+    this._signable.set(ns, 0)
+    this._hash = this._signable.subarray(32)
+  }
+
+  sign (payload, secretKey, out = b4a.allocUnsafe(64)) {
+    sodium.crypto_generichash(this._hash, payload)
+    sodium.crypto_sign_detached(out, this._signable, secretKey)
```
Comment (mafintosh) [inline lib/namespace.js:15]: sign([ns, signable])

Comment (mafintosh) [reply lib/namespace.js:15]: with no hashing step

**Comment by kasperisager** [comment]
@mafintosh Let me know if 5dbf4b557e0eee2d0c2cd23d25fa3ff10d37933c is correct.

**Comment by mafintosh** [comment]
Lets close for now and revisit later if needed

**Pull request closed without merging**